neo3/wallet/account.py

```diff
@@ -0,0 +1,283 @@
+
+        Returns:
+            The account that will be monitored.
+        """
+        return cls(password='', watch_only=True, address=cls.script_hash_to_address(script_hash))
+
+    @classmethod
+    def watch_only_from_address(cls, address: str) -> Account:
+        """
+        Instantiate and returns a watch-only account from a given address.
+
+        Args:
+            address: the address that will identify an account to be watched.
+
+        Returns:
+            The account that will be monitored.
+        """
+        return cls(password='', watch_only=True, address=address)
+
+    @staticmethod
+    def script_hash_to_address(script_hash: types.UInt160) -> str:
+        """
+        Converts the specified script hash to an address.
+
+        Args:
+            script_hash: script hash to convert.
+        """
+        version = settings.network.account_version  # this is the current Neo's protocol version
+        data_ = version.to_bytes(1, 'little') + script_hash.to_array()
+
+        return base58.b58encode_check(data_).decode('utf-8')
+
+    @staticmethod
+    def address_to_script_hash(address: str) -> types.UInt160:
+        """
+        Converts the specified address to a script hash.
+
+        Args:
+            address: address to convert
+
+        Raises:
+            ValueError: if the length of data_ is not valid.
+            ValueError: if the script hash version is not valid.
+        """
+        data_ = base58.b58decode_check(address)
+        if not Account.is_valid_address(address):
+            if len(data_) != len(types.UInt160.zero()) + 1:
+                raise ValueError(f"The address is wrong, because data_ length should be "
+                                 f"{len(types.UInt160.zero()) + 1}")
+
+            if data_[0] != settings.network.account_version:
+                raise ValueError(f"The account version is not {settings.network.account_version}")
+
+        return types.UInt160(data=data_[1:])
+
+    @staticmethod
+    def private_key_from_nep2(nep2_key: str, passphrase: str) -> bytes:
+        """
+        Decrypt a nep2 key into a private key.
+
+        Args:
+            nep2_key: the key that will be decrypt.
+            passphrase: the password to decrypt the nep2 key.
+
+        Raises:
+            ValueError: if the length of the nep2_key is not valid.
+            ValueError: if it's not possible to decode the nep2_key.
+            ValueError: if the passphrase is incorrect or the version of the account is not valid.
+
+        Returns:
+            the private key.
+        """
+        if len(nep2_key) != 58:
+            raise ValueError(f"Please provide a nep2_key with a length of 58 bytes (LEN: {len(nep2_key)})")
+
+        address_hash_size = 4
+        address_hash_offset = len(NEP_FLAG) + len(NEP_HEADER)
+
+        try:
+            decoded_key = base58.b58decode_check(nep2_key)
+        except Exception:
+            raise ValueError("Invalid nep2_key")
```

Comment:
```suggestion
            raise ValueError("Base58decode failure of nep2 key")
```
The existing message is a little too broad to identify what is wrong with it. The other 2 exceptions give a specific direction the user can look into, this one didn't